docs/tutorials/tutorial_2.md

```diff
@@ -0,0 +1,177 @@
+    apt-get upgrade -y && \
+    apt-get install git -y
+
+# Install requirements
+COPY requirements.txt /
+RUN pip3 install --no-cache-dir -r requirements.txt
+
+# Install Fondant
+# This is split from other requirements to leverage caching
+ARG FONDANT_VERSION=main
+RUN pip3 install fondant[aws,azure,gcp]@git+https://github.com/ml6team/fondant@main
+
+# Set the working directory to the component folder
+WORKDIR /component/src
+
+# Copy over src-files
+COPY src/ .
+
+ENTRYPOINT ["fondant", "execute", "main"]
+```
+
+As part of the Dockerfile build process, we install necessary dependencies. Consequently, we must create a `requirements.txt` file in the `components/filter_images` folder. If your components logic demands custom libraries, you can include them in the requirements.txt file.
+
+
+## Make use of your component
+
+To utilise your component, you can incorporate it into the pipeline created in [Tutorial 1](tutorials/tutorial_1.md). To do this, you'll need to add the following code to the `pipeline.py` file:
+
+```python
+# Filter mime type component
+filter_mime_type = ComponentOp(
+    component_dir="components/filter_file_type",
+    arguments={"mime_type": "image/png"}
+)
+```
+
+We initialise the component from a local path, similar to the generic component. However, in this case, the component will be built entirely based on your local files, as the folders contain additional information beyond the ComponentSpec.
+
+Lastly, we need to make adjustments to the pipeline. The step for downloading images can be network-intensive since it involves actual downloads. As a result, we want to pre-filter the files before proceeding with the downloads. To achieve this, we'll modify the pipeline as follows:
+
+```python
+pipeline.add_op(load_from_hf_hub)
+pipeline.add_op(filter_mime_type, dependencies=[load_from_hf_hub])
+pipeline.add_op(download_images, dependencies=[filter_mime_type])
+```
+
+We are inserting our custom component as an intermediary step within our pipeline.
+
+Now, you can execute the pipeline once more and examine the results. The final output should exclusively consist of PNG images.
+
+We have designed the custom component to be easily adaptable. For example, if you wish to filter out JPEG files, you can simply change the argument to `image/jpeg`, and your dataset will be populated with JPEGs instead of PNGs
+
+## You Are Done! 🎉 
+We now have a simple pipeline that downloads a dataset from HuggingFace hub and outputs the dataset filtered by PNG images. A possible next step is to create a component that **filters the data based on the aspect ratio**? Or **run a CLIP model on the images to get embeddings**?
+
+Expanding upon the concept of custom component implementation, you have the flexibility to create additional custom components. For example, you can design custom components to filter out NSFW (Not Safe For Work) content or to identify and exclude images containing watermarks.
+
+## Some things to pay attention to
```

Comment:
this section was a little + but I don't think it belongs to the tutorial_2. It's a general advice for using fondant, not only for the tutorial_2. Maybe it can be elsewhere like in the getting_started section or so? Or we can just not put it anywhere. 
My feeling is that it could be mentioned next to prerequisites or so.